From 74a1db4edfa8af9ea5e59f4a16775a09f31497eb Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:45:16 -0400
Subject: [x86] pci: Remove unnecessary list_empty(&pci_root_infos) check

Message-id: <20140806034515.1438.99284.stgit@gir.stowe>
Patchwork-id: 87420
O-Subject: [RHEL7.1 PATCH 11/69] x86/PCI: Remove unnecessary list_empty(&pci_root_infos) check
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 3323ab8f7a2f8ffed9393388863bf17329077524

commit 3323ab8f7a2f8ffed9393388863bf17329077524
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Fri Jan 24 11:57:18 2014 -0700

    x86/PCI: Remove unnecessary list_empty(&pci_root_infos) check

    list_for_each_entry() handles empty lists, so there's no need to check
    whether the list is empty first.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/pci/bus_numa.c b/arch/x86/pci/bus_numa.c
index 2e36a44..f3a2cfc 100644
--- a/arch/x86/pci/bus_numa.c
+++ b/arch/x86/pci/bus_numa.c
@@ -10,9 +10,6 @@ static struct pci_root_info *x86_find_pci_root_info(int bus)
 {
  struct pci_root_info *info;
 
- if (list_empty(&pci_root_infos))
-  return NULL;
-
  list_for_each_entry(info, &pci_root_infos, list)
   if (info->busn.start == bus)
    return info;
-- 
1.7.1